File: csharp/Behavioral/16-Mediator/Colleague/Handyman.cs
public class Handyman : Colleague
{
    public string Specialisation { get; }

    public Handyman(IDispatchMediator mediator, string name, string specialisation)
        : base(mediator, name)
    {
        Specialisation = specialisation;
    }

    public void CompleteJob(string jobId)
    {
        Console.WriteLine($"\n[{Name}] Marking Job #{jobId} as complete.");
        Mediator.Notify(this, "JobCompleted", new { JobId = jobId, Handyman = Name });
    }

    public void ReceiveAssignment(string jobTitle, string slot)
    {
        Console.WriteLine($"  [{Name}] Assigned: '{jobTitle}' at {slot}");
    }
}

File: typescript/Behavioral/16-Mediator/Colleague/Handyman.ts
import { Colleague } from "./Colleague";
import { IDispatchMediator } from "../Mediator/IDispatchMediator";

export class Handyman extends Colleague {
  constructor(
    mediator: IDispatchMediator,
    name: string,
    readonly specialisation: string,
  ) {
    super(mediator, name);
  }

  completeJob(jobId: string): void {
    console.log(`\n[${this.name}] Marking Job #${jobId} as complete.`);
    this.mediator.notify(this, "JobCompleted", { jobId, handyman: this.name });
  }

  receiveAssignment(jobTitle: string, slot: string): void {
    console.log(`  [${this.name}] Assigned: '${jobTitle}' at ${slot}`);
  }
}
